tests/cpp/helpers.cc

```diff
@@ -478,4 +487,57 @@ std::unique_ptr<GradientBooster> CreateTrainedGBM(
   return gbm;
 }
 
+#if defined(XGBOOST_USE_RMM) && XGBOOST_USE_RMM == 1
+
+using cuda_mr_t = rmm::mr::cuda_memory_resource;
+using pool_mr_t = rmm::mr::pool_memory_resource<cuda_mr_t>;
+class RMMAllocator {
+ public:
+  std::vector<std::unique_ptr<cuda_mr_t>> cuda_mr;
+  std::vector<std::unique_ptr<pool_mr_t>> pool_mr;
+  int n_gpu;
+  RMMAllocator() : n_gpu(common::AllVisibleGPUs()) {
+    int current_device;
+    CHECK_EQ(cudaGetDevice(&current_device), cudaSuccess);
+    for (int i = 0; i < n_gpu; ++i) {
+      CHECK_EQ(cudaSetDevice(i), cudaSuccess);
+      cuda_mr.push_back(std::unique_ptr<cuda_mr_t>(new cuda_mr_t));
+      pool_mr.push_back(std::unique_ptr<pool_mr_t>(new pool_mr_t(cuda_mr[i].get())));
+    }
+    CHECK_EQ(cudaSetDevice(current_device), cudaSuccess);
```

Comment:
`dh::safe_cuda`.

Reply:
Not available here, since this file has `.cc` extension.